website/docs/commands/cli-config.html.markdown

```diff
@@ -144,14 +148,199 @@ To learn how to write and install your own credentials helpers to integrate
 with existing in-house credentials management systems, see
 [the guide to Credentials Helper internals](/docs/internals/credentials-helpers.html).
 
-## Deprecated Settings
+## Provider Installation
+
+The default way to install provider plugins is from a provider registry. The
+origin registry for a provider is encoded in the provider's source address,
+like `registry.terraform.io/hashicorp/aws`. For convenience in the common case,
+Terraform allows omitting the hostname portion for providers on
+`registry.terraform.io`, so we'd normally write `hashicorp/aws` instead in
+this case.
+
+Downloading a plugin directly from its origin registry is not always
+appropriate, though. For example, the system where you are running Terraform
+may not be able to access an origin registry due to firewall restrictions
+within your organization or your locality.
+
+To allow using Terraform in these situations, there are some alternative
```

Comment:
tiny nitpick (and totally ok if you don't agree): it's more specific and accurate to say this features allows installing/using terraform _providers_ in these situations

Reply:
Fair! I think I was assuming that Terraform is useless if you can't install providers, but no need to be vague about it.